pkg/idp/idp.go

```diff
@@ -85,29 +86,41 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	if err := s.Get(ctx, types.NamespacedName{Name: dexSecretConfig.Name, Namespace: dexSecretConfig.Namespace}, secret); err != nil {
 		if !apierrors.IsNotFound(err) {
 			return microerror.Mask(err)
-		} else {
-
-			// Create apps for each provider and get dex config
-			appConfig, err := s.GetAppConfig(ctx)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-			dexConfig, err := s.CreateProviderApps(appConfig, ctx)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-			data, err := json.Marshal(dexConfig)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-			// Create secret
-			secret = s.GetDefaultDexConfigSecret(dexSecretConfig.Name, dexSecretConfig.Namespace, data)
+		}
+	}
+	{
+		// Get existing connectors from the dex config secret
+		oldConnectors, err := getConnectorsFromSecret(secret)
```

Comment:
We save the connectors that were previously configured in a map so that we can later compare what was changed.